File: compiler-group-traslator/src/test/resources/c/ForwardClassTest.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>
#include "GC.c"
#include "ExceptionHandlingFunctions.c"
VM* vm;
typedef struct Foo2 {
	struct Bar* b;
}Foo2;
typedef struct Bar {
	int x;
}Bar;

int main(char* args) {
	vm = newVM();
	struct Foo2* a;

	a = (Foo2*)newObject(vm, &a, sizeof (Foo2)) ;

	a->b = (Bar*)newObject(vm, &a, sizeof (Bar)) ;

	a->b->x = 5 ;
	return 0;
}

File: compiler-group-traslator/src/test/resources/ForwardClassTest.java

public class ForwardClassTest
{
    public static void main(String[] args)
    {
        Foo2 a;
        
        a = new Foo2();
        a.b = new Bar();
        a.b.x = 5;
    }
}

class Foo2
{
    public Bar b;
}

class Bar
{
    public int x;
}
